### PR TITLE
Feature/scat 2136 projects list

### PR DESCRIPTION
**Pull request by sudhakar-reddy-konda**
### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java [hunk 1]
```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
...
+   * @param project the project
+   * @return ProjectPackageSummary
+   */
+  private ProjectPackageSummary convertProjectToProjectPackageSummary(Project project) {
+	  // Get Project from database
+	  var dbProject = retryableTendersDBDelegate
+			  .findProcurementExternalProjectById(project.getTender().getTenderCode());
+	  var projectPackageSummary = new ProjectPackageSummary();
+	  if (dbProject.isPresent()) {
+		  var dbEvent = getCurrentEvent(dbProject.get());
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:361]: I get a `500` with msg `Could not find current event for project 13` when I invoke the endpoint.  This is due to me having a lot of old projects in Jaggaer, at least one of which doesn't have any events - may be an edge case but probably need to trap it.

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java [hunk 2]
```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
...
+
+		  //TODO no value for Uri
+		  //projectPackageSummary.setUri(getProjectUri);
+		  projectPackageSummary.setProjectId(dbProject.get().getId());
+		  projectPackageSummary.setProjectName(dbProject.get().getProjectName());
+		  var eventSummary = new EventSummary();
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:367]: There is a `TendersAPIModelUtils.buildEventSummary(...)` method that could be reused here..

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java [hunk 3]
```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
...
+		  eventSummary.setId(dbEvent.getEventID());
+		  eventSummary.setTitle(dbEvent.getEventName());
+		  //TODO No value for event stage
+		  //eventSummary.setEventStage(ReleaseTag.fromValue(dbEvent.getEventStage()));
+		  eventSummary.setEventType(ViewEventType.fromValue(dbEvent.getEventType()));
+		  projectPackageSummary.activeEvent(eventSummary);
+		  return projectPackageSummary;
+	  } else {
+		  // TODO ignoring at the moment. What to do with these?
+		  log.warn("Unable to project details in database for project id  {}, so ignoring.",
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:377]: Does it say anything on the JIRA about what to do with existing projects in Jaggaer that don't exist in Tenders DB (i.e. have been created by the user before onboarding to CaT Buyer UI)?  If not, then this is probably the only sensible course of action (only other thing we could do it create the DB entries from the 'missing' projects at this point..) 🤔

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java [hunk 4]
```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
...
+				  project.getTender().getTenderCode());
+	  }
+	  return null;
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:380]: Could return `Optional.empty()` then filter on `Optional::isPresent` - might be slightly more robust.

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java [hunk 5]
```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
...
+  }
+
+  /**
+   * convert project top project package summary
+   * @param project the project
+   * @return ProjectPackageSummary
+   */
+  private ProjectPackageSummary convertProjectToProjectPackageSummary(Project project) {
+	  // Get Project from database
+	  var dbProject = retryableTendersDBDelegate
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:357]: For me, again not representing reality as I have loads of old projects in Jaggaer, this results in a lot od individual calls to the DB (one for each project in Jaggaer).  Suggest perhaps changing the query to `findByExternalProjectIdIn(Set<String> externalProjectIds)` - that should get a collection and then could use that to do the compare between what is in Jaggaer and what is in the DB perhaps?

**Review by tommyb82 — COMMENTED** [review]
Looking good, just a few suggestions / comments 👍

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/jaggaer/ProjectList.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.extern.jackson.Jacksonized;
+
+import java.util.List;
+
+@Value
+@Builder
+@Jacksonized
+public class ProjectList {
+
+  private List<Project> project;
```
[the remark this answers is not in the document]
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/jaggaer/ProjectList.java]: Done

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/ProjectsController.java [hunk 7]
```diff
@@ -24,14 +24,21 @@
 
   private final ProcurementProjectService procurementProjectService;
 
+  @GetMapping("/")
+  public Collection<ProjectPackageSummary> getProjects(final JwtAuthenticationToken authentication){
+    String principal = getPrincipalFromJwt(authentication);
```
[the remark this answers is not in the document]
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/ProjectsController.java]: done

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java [hunk 8]
```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
...
+    var jaggaerUserId = userProfileService.resolveBuyerUserByEmail(principal)
+            .orElseThrow(() -> new AuthorisationFailureException("Jaggaer user not found")).getUserId();
+    ProjectListResponse projectListResponse = jaggaerService.getProjectList(jaggaerUserId);
+    if (!CollectionUtils.isEmpty(projectListResponse.getProjectList().getProject())) {
+      return projectListResponse.getProjectList().getProject().stream()
+              .map(this::convertProjectToProjectPackageSummary)
+              .filter(projectPackageSummary -> projectPackageSummary != null) // TODO any better?
+              .collect(Collectors.toList());
+    }
+    return Collections.EMPTY_LIST;
```
[the remark this answers is not in the document]
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java]: done

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java (hunk 2 above)
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java]: done

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java (hunk 4 above)
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java]: done

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java (hunk 3 above)
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:381]: Update JIRA ticket with question. will leave it as it is at the moment

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java (hunk 1 above)
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java:363]: Update JIRA ticket with question. will leave it as it is at the moment

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java (hunk 5 above)
Comment (sudhakar-reddy-konda) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java]: done

**Review by tommyb82 — APPROVED** [review]
👍  Great work

**Pull request merged**